plugins/catalog-react/src/alpha/blueprints/OverviewEntityContentLauyoutBlueprint.tsx

```diff
@@ -0,0 +1,137 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import {
+  createExtensionDataRef,
+  createExtensionBlueprint,
+  ExtensionBoundary,
+} from '@backstage/frontend-plugin-api';
+import React, { lazy as reactLazy, ComponentProps } from 'react';
+import { EntityCardBlueprint } from './EntityCardBlueprint';
+
+/** @alpha */
+export interface OverviewEntityContentLayoutProps {
+  buildFilterFn: (
+    filterFunction?: (entity: Entity) => boolean,
+    filterExpression?: string,
+  ) => (entity: Entity) => boolean;
+  cards: Array<{
+    area?: string;
+    element: React.JSX.Element;
+    filterFunction?: (entity: Entity) => boolean;
+    filterExpression?: string;
+  }>;
+}
+
+const catalogOverviewEntityContentLayoutAreasDataRef = createExtensionDataRef<
+  Array<string>
+>().with({
+  id: 'catalog.overview-entity-content.layout.areas',
+});
+
+const catalogOverviewEntityContentLayoutFilterFunctionDataRef =
+  createExtensionDataRef<(entity: Entity) => boolean>().with({
+    id: 'catalog.overview-entity-content.layout.filter-function',
+  });
+
+const catalogOverviewEntityContentLayoutFilterExpressionDataRef =
+  createExtensionDataRef<string>().with({
+    id: 'catalog.overview-entity-content.layout.filter-expression',
+  });
+
+const catalogOverviewEntityContentLayoutComponentDataRef =
+  createExtensionDataRef<
+    (props: OverviewEntityContentLayoutProps) => React.JSX.Element
+  >().with({
+    id: 'catalog.overview-entity-content.layout.page-component',
+  });
+
+/** @alpha */
+export const OverviewEntityContentLauyoutBlueprint = createExtensionBlueprint({
+  kind: 'catalog-overview-entity-content-layout',
+  attachTo: { id: 'entity-content:catalog/overview', input: 'layouts' },
+  output: [
+    catalogOverviewEntityContentLayoutAreasDataRef,
+    EntityCardBlueprint.dataRefs.area,
+    catalogOverviewEntityContentLayoutFilterFunctionDataRef.optional(),
+    catalogOverviewEntityContentLayoutFilterExpressionDataRef.optional(),
+    catalogOverviewEntityContentLayoutComponentDataRef,
+  ],
+  dataRefs: {
+    areas: catalogOverviewEntityContentLayoutAreasDataRef,
+    defaultArea: EntityCardBlueprint.dataRefs.area,
+    filterFunction: catalogOverviewEntityContentLayoutFilterFunctionDataRef,
+    filterExpression: catalogOverviewEntityContentLayoutFilterExpressionDataRef,
+    component: catalogOverviewEntityContentLayoutComponentDataRef,
+  },
+  config: {
+    schema: {
+      area: z => z.string().optional(),
+      filter: z => z.string().optional(),
+    },
+  },
+  *factory(
+    {
+      loader,
+      areas,
+      defaultArea,
+      defaultFilter = () => false, // do not use the layout if a default filter is undefined
+    }: {
+      areas: Array<string>;
+      defaultArea: (typeof areas)[number]; // fixed typescript area type
```

Comment:
@Rugvip I tried inferring the type of the default area from the string values specified in the areas array, but I was not able to. Is it possible?

Reply:
Not necessary anymore